vendor_data.py

```diff
@@ -56,101 +52,190 @@ def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "X
         else:
-            with pd.ExcelWriter(f"{filename}.xlsx", mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+            with pd.ExcelWriter(filename, mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
                 df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)
 
-    def change_status(self):
-        self.Preferred = False
-        if self.Preferred == True:
-            self.Status = "Preferred"
-        else:
-            self.Status = "Back-Up"
-
-
-
 # don't forget to properly dcument them with docstrings!!
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser("docstring")
-    parser.add_argument("VendorName", type=str, help="Vendor's name")
-    parser.add_argument("TargLang", type=str, help="Language the vendor will be translating into")
-    parser.add_argument("--WordRate", type=float, help="The vendor's word rate in Euro")
-    parser.add_argument("--VendorMail", type=str, help="The vendor's email address")
-    parser.add_argument("--CatTool", type=str, help="The CAT-tool the vendor wants to use")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-a", "--add", action='store_true', help = "Add a vendor")
+    parser.add_argument ("-m", "--modify", action='store_true', help = "Modify an existing vendor")
     args = parser.parse_args()
-    Vndr = VendorData(args.VendorName, args.TargLang, args.WordRate, args.VendorMail, args.CatTool)
+    
+    if args.add == True:
+        done = "no"
+        WordRate = None
+        Preferred = None
+        VendorMail = ""
+        CatTool = "XTM"
+        while done.lower() != "yes":
+            ProjectName = pyip.inputStr("What is the project name? ")
+            SourceLang = pyip.inputStr("What is the source language? ")
+            VendorName = pyip.inputStr("What's the vendor's name? ")
+            TargLang = pyip.inputStr("Into which language will the vendor translate? ")
+            NewWordRate = pyip.inputNum("What is the vendor's word rate in EUR? Should be between 0.01 and 0.15. If higher, choose another vendor. ", blank = True)
+            VendorWordRate = NewWordRate if NewWordRate else WordRate
+            NewPreferred = pyip.inputBool("True or False: Is this vendor a preferred vendor? If neither, leave blank. ", blank = True, default=None)
+            Preferred = NewPreferred if NewPreferred else Preferred
+            NewVendorMail = pyip.inputStr("What is the vendor's email address? ", blank = True, default="")
+            VendorMail = NewVendorMail if NewVendorMail else VendorMail
+            NewCatTool= pyip.inputMenu(["XTM", "Trados Studio", "MemoQ", "Memsource"], prompt = "In which tool will the vendor be working?", blank = True, default = "XTM")
+            CatTool = NewCatTool if NewCatTool else CatTool
+            done = pyip.inputStr("Are you done? ")
+            if done.lower() == "yes":

```

Comment:
Everything that you wrote before will be run as long as `done.lower()` is not equal to "yes". So you don't need this line. Just align lines 196-196 to align with line 178 and they will run when the while loop is done, which is when `done.lower()` becoems "yes" :)